pkg/admin/cache/registry/universal/cache.go

```diff
@@ -0,0 +1,474 @@
+						Application: &cache.ApplicationModel{Name: application},
+						Workload:    nil,
+						Name:        serviceKey + "#" + dubboModel.Ip + ":" + dubboModel.Port,
+						Ip:          dubboModel.Ip,
+						Port:        dubboModel.Port,
+						Status:      "",
+						Node:        "",
+						Labels:      nil,
+					})
+				}
+			}
+		}
+	}
+	uc.consumers.lock.RUnlock()
+
+	return res, nil
+}
+
+func (uc *UniversalCache) GetInstancesWithSelector(namespace string, selector selector.Selector) ([]*cache.InstanceModel, error) {
+	res := make([]*cache.InstanceModel, 0)
+	instanceSet := map[string]struct{}{}
+
+	uc.providers.lock.RLock()
+	for application, serviceMap := range uc.providers.data {
+		if targetApplication, ok := selector.ApplicationOption(); ok && targetApplication != application {
+			continue
+		} else {
+			for serviceKey, instanceMap := range serviceMap {
+				for _, dubboModel := range instanceMap {
+					if _, ok := instanceSet[dubboModel.Ip+":"+dubboModel.Port]; ok {
+						continue
+					} else {
+						instanceSet[dubboModel.Ip+":"+dubboModel.Port] = struct{}{}
+						res = append(res, &cache.InstanceModel{
+							Application: &cache.ApplicationModel{Name: application},
+							Workload:    nil,
+							Name:        serviceKey + "#" + dubboModel.Ip + ":" + dubboModel.Port,
+							Ip:          dubboModel.Ip,
+							Port:        dubboModel.Port,
+							Status:      "",
+							Node:        "",
+							Labels:      nil,
+						})
+					}
+				}
+			}
+		}
+	}
+	uc.providers.lock.RUnlock()
+
+	uc.consumers.lock.RLock()
+	for application, serviceMap := range uc.consumers.data {
+		if targetApplication, ok := selector.ApplicationOption(); ok && targetApplication != application {
+			continue
+		} else {
+			for serviceKey, instanceMap := range serviceMap {
+				for _, dubboModel := range instanceMap {
+					if _, ok := instanceSet[dubboModel.Ip+":"+dubboModel.Port]; ok {
+						continue
+					} else {
+						instanceSet[dubboModel.Ip+":"+dubboModel.Port] = struct{}{}
+						res = append(res, &cache.InstanceModel{
+							Application: &cache.ApplicationModel{Name: application},
+							Workload:    nil,
+							Name:        serviceKey + "#" + dubboModel.Ip + ":" + dubboModel.Port,
+							Ip:          dubboModel.Ip,
+							Port:        dubboModel.Port,
+							Status:      "",
+							Node:        "",
+							Labels:      nil,
+						})
+					}
+				}
+			}
+		}
+	}
+	uc.consumers.lock.RUnlock()
+
+	return res, nil
+}
+
+func (uc *UniversalCache) GetServices(namespace string) ([]*cache.ServiceModel, error) {
+	res := make([]*cache.ServiceModel, 0)
+
+	uc.providers.lock.RLock()
+	for application, serviceMap := range uc.providers.data {
+		for serviceKey := range serviceMap {
+			res = append(res, &cache.ServiceModel{
+				Application: &cache.ApplicationModel{Name: application},
+				Category:    constant.ProviderSide,
+				Name:        util.GetInterface(serviceKey),
+				Labels:      nil,
```

Comment:
Consider putting URL params into Labels.

Reply:
The labels filed only represent labels under kubernetes, maybe we can add a field to represent the URL's params like params?

Reply:
I think maybe better if we can define a unified concept, for example `Labels`, if so, we can put Kubernets labels or URL params into that unified concept.